pkg/controller/sparkapplication/sparkapp_metrics.go

```diff
@@ -196,6 +218,18 @@ func (sm *sparkAppMetrics) exportMetrics(oldApp, newApp *v1beta2.SparkApplicatio
 		}
 	}
 
+	// In the event that state transitions happened too quickly and the spark app skipped the RUNNING state, the job
+	// start latency should still be captured.
+	// Note: There is an edge case that a Submitted state can go directly to a Failing state if the driver pod is
+	//       deleted. This is very unlikely if not being done intentionally, so we choose not to handle it.
+	if newState != oldState {
+		if (newState == v1beta2.FailingState || newState == v1beta2.SucceedingState) && oldState != v1beta2.RunningState {
+			// TODO: remove this log once we've gathered some data in prod fleets.
+			glog.Infof("Calculating job start latency metrics for edge case transition from %v to %v in app %v in namespace %v.", oldState, newState, newApp.Name, newApp.Namespace)
```

Comment:
Can we change the level to `2` for this log?

Reply:
Sure